qiskit/transpiler/passes/optimization/commutative_cancellation.py

```diff
@@ -51,6 +69,18 @@ def run(self, dag):
         Raises:
             TranspilerError: when the 1-qubit rotation gates are not found
         """
+        var_z_gate = None
+        z_var_gates = [gate for gate in dag.properties()['operations'].keys()
+                       if gate in set(self._z_variable_gates)]
+        if z_var_gates:
+            # priortize z gates in circuit
+            var_z_gate = self._var_z_map[next(iter(z_var_gates))]
+        else:
+            z_var_gates = [gate for gate in self.basis
+                           if gate in set(self._z_variable_gates)]
+            if z_var_gates:
+                var_z_gate = self._var_z_map[next(iter(z_var_gates))]
+
```

Comment:
What will happen if no `var_z_gate` is found, either from the circuit or from `basis_gates`? Should the pass return early?

Reply:
If not `var_z_gate` is found no cancellation is attempted. I moved this check earlier.